src/Form/selectors.js

```diff
@@ -76,35 +100,52 @@ const CC_EXP_SELECTOR = `
 input[id*=expiration i],
 select[id*=expiration i]`
 
-/* This is used to map a selector with the data type we store for credit cards */
-const CC_SELECTORS_MAP = {
-    [CC_NAME_SELECTOR]: {
-        ccType: 'cardName',
+// Matches strings like mm/yy, mm-yyyy, mm-aa
+const DATE_SEPARATOR_REGEX = /\w\w\s?(?<separator>[/\s.\-_—–])\s?\w\w/i
+// Matches 4 non-digit repeated characters (YYYY or AAAA) or 4 digits (2022)
+const FOUR_DIGIT_YEAR_REGEX = /(\D)\1{3}|\d{4}/i
+
+/**
+ * This is used to map a selector with the data type we store for credit cards
+ * @type {[Matcher]}
+ */
+const CC_MATCHERS_LIST = [
+    {
+        type: 'cardName',
+        selector: CC_NAME_SELECTOR,
         regex: /(card.*name|name.*card)|(card.*holder|holder.*card)|(card.*owner|owner.*card)/i
     },
-    [CC_NUMBER_SELECTOR]: {
-        ccType: 'cardNumber',
+    {
+        type: 'cardNumber',
+        selector: CC_NUMBER_SELECTOR,
         regex: /card.*number|number.*card/i
     },
-    [CC_CVC_SELECTOR]: {
-        ccType: 'cardSecurityCode',
+    {
+        type: 'cardSecurityCode',
+        selector: CC_CVC_SELECTOR,
         regex: /security.?code|cvv|csc|cvc/i
     },
-    [CC_MONTH_SELECTOR]: {
-        ccType: 'expirationMonth',
-        regex: /(card|cc)?.?(exp(iry|iration)?)?.?(mo(nth)?|mm)/i
+    {
+        type: 'expirationMonth',
+        selector: CC_MONTH_SELECTOR,
+        regex: /(card|cc)?.?(exp(iry|iration)?)?.?(month|mm(?![.\s/-]yy))/i,
+        negativeRegex: /mm[/\s.\-_—–]/i
     },
-    [CC_YEAR_SELECTOR]: {
-        ccType: 'expirationYear',
-        regex: /(card|cc)?.?(exp(iry|iration)?)?.?(ye(ar)?|yy)/i
+    {
+        type: 'expirationYear',
+        selector: CC_YEAR_SELECTOR,
+        regex: /(card|cc)?.?(exp(iry|iration)?)?.?(ye(ar)?|yy)/i,
+        negativeRegex: /mm[/\s.\-_—–]/i
     },
-    [CC_EXP_SELECTOR]: {
-        ccType: 'expiration',
-        regex: /exp(iry|iration)?/i
+    {
+        type: 'expiration',
+        selector: CC_EXP_SELECTOR,
+        regex: /(mm|\d\d)[/\s.\-_—–](yy|jj|aa|\d\d)|exp|valid/i,
+        negativeRegex: /invalid/i
```

Comment:
Safari and Webkit don't support negative lookbehinds, so I had to use `negativeRegex`.